ADAL/src/ADLogger+Internal.h

```diff
@@ -23,52 +23,22 @@
-    [ADLogger log:_level context:self message:_msg errorCode:_code correlationId:_correlationId userInfo:_dict format:_fmt, ##__VA_ARGS__ ]
-
-#define AD_LOG_ERROR(_message, _code, _correlationId, _info) \
-    AD_LOG(ADAL_LOG_LEVEL_ERROR, (_message), (_code), _correlationId, nil, _info)
-#define AD_LOG_WARN(_message, _correlationId, _info) \
-    AD_LOG(ADAL_LOG_LEVEL_WARN, _message, AD_ERROR_SUCCEEDED, _correlationId, nil, _info)
-#define AD_LOG_INFO(_message, _correlationId, _info) \
-    AD_LOG(ADAL_LOG_LEVEL_INFO, _message, AD_ERROR_SUCCEEDED, _correlationId, nil, _info)
-#define AD_LOG_VERBOSE(_message, _correlationId, _info) \
-    AD_LOG(ADAL_LOG_LEVEL_VERBOSE, _message, AD_ERROR_SUCCEEDED, _correlationId, nil, _info)
-
-#define AD_LOG_ERROR_F(_msg, _code, _correlationId, _fmt, ...) \
-    AD_LOG_F(ADAL_LOG_LEVEL_ERROR, _msg, _code, _correlationId, nil, _fmt, ##__VA_ARGS__)
-#define AD_LOG_WARN_F(_msg, _correlationId, _fmt, ...) \
-    AD_LOG_F(ADAL_LOG_LEVEL_WARN, _msg, AD_ERROR_SUCCEEDED, _correlationId, nil, _fmt, ##__VA_ARGS__)
-#define AD_LOG_INFO_F(_msg, _correlationId, _fmt, ...) \
-    AD_LOG_F(ADAL_LOG_LEVEL_INFO, _msg, AD_ERROR_SUCCEEDED, _correlationId, nil, _fmt, ##__VA_ARGS__)
-#define AD_LOG_VERBOSE_F(_msg, _correlationId, _fmt, ...) \
-    AD_LOG_F(ADAL_LOG_LEVEL_VERBOSE, _msg, AD_ERROR_SUCCEEDED, _correlationId, nil, _fmt, ##__VA_ARGS__)
-
-#define AD_LOG_ERROR_DICT(_message, _code, _correlationId, _dict, _info) \
-    AD_LOG(ADAL_LOG_LEVEL_ERROR, (_message), (_code), _correlationId, _dict, _info)
-#define AD_LOG_WARN_DICT(_message, _correlationId, _dict, _info) \
-    AD_LOG(ADAL_LOG_LEVEL_WARN, _message, AD_ERROR_SUCCEEDED, _correlationId, _dict, _info)
-#define AD_LOG_INFO_DICT(_message, _correlationId, _dict, _info) \
-    AD_LOG(ADAL_LOG_LEVEL_INFO, _message, AD_ERROR_SUCCEEDED, _correlationId, _dict, _info)
-#define AD_LOG_VERBOSE_DICT(_message, _correlationId, _dict, _info) \
-    AD_LOG(ADAL_LOG_LEVEL_VERBOSE, _message, AD_ERROR_SUCCEEDED, _correlationId, _dict, _info)
-
-#define AD_LOG_ERROR_DICT_F(_msg, _code, _correlationId, _dict, _fmt, ...) \
-    AD_LOG_F(ADAL_LOG_LEVEL_ERROR, _msg, _code, _correlationId, _dict, _fmt, ##__VA_ARGS__)
-#define AD_LOG_WARN_DICT_F(_msg, _correlationId, _dict, _fmt, ...) \
-    AD_LOG_F(ADAL_LOG_LEVEL_WARN, _msg, AD_ERROR_SUCCEEDED, _correlationId, _dict, _fmt, ##__VA_ARGS__)
-#define AD_LOG_INFO_DICT_F(_msg, _correlationId, _dict, _fmt, ...) \
-    AD_LOG_F(ADAL_LOG_LEVEL_INFO, _msg, AD_ERROR_SUCCEEDED, _correlationId, _dict, _fmt, ##__VA_ARGS__)
-#define AD_LOG_VERBOSE_DICT_F(_msg, _correlationId, _dict, _fmt, ...) \
-    AD_LOG_F(ADAL_LOG_LEVEL_VERBOSE, _msg, AD_ERROR_SUCCEEDED, _correlationId, _dict, _fmt, ##__VA_ARGS__)
+#define AD_LOG_ERROR(_code, _correlationId, _dict, _isPii, _fmt, ...) \
+    AD_LOG(ADAL_LOG_LEVEL_ERROR, _code, _correlationId, _dict, _isPii, _fmt, ##__VA_ARGS__)
+
+#define AD_LOG_WARN(_correlationId, _dict, _isPii, _fmt, ...) \
```

Comment:
If we're going to change these macros that needs some wider design discussion, my preference here would be to:

1. Remove the userInfo parameter entirely. This was a weird hacky way to put in telemetry without really doing telemetry. We have a real telemetry channel now, I'd rather just get rid of this.
2. Instead of isPII being a parameter make separate _PII macros

Reply:
Also, remove error code from the error logging macros, there's plenty of cases where we want to log error information that's not tied to a specific error code and that's led to teams in the past filing issues based on error log statements rather then actual end result errors.

Reply:
* I agree with number 1 (removing the userInfo and the error code), as it doesn't seem to provide much value. Unless there're good historical reasons to keep it around, I totally support removing them.

* I don't completely agree with having separate _PII macros. From a developer perspective, if I forced to pass the PII flag when using the macro, it forces me to think if it has PII or not. If there's a separate _PII macro instead, I can very well just use the normal macro and miss PII. I'm open to discussion on this one, but that's my personal opinion :)

Reply:
I agree with Olga's point regarding separate/one macro(s). I think it would be safer to force developer to pass the PII flag.

Reply:
I agree with (1) and about removing error code. 
(2), I am open to both; but if we have a separate macro, it will read better in terms of reviews and all vs having a boolean value as one of the parameters.